examples/threading/threading.c

```diff
@@ -13,8 +13,13 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
-    return thread_param;
+    thread_data_t* thread_func_args = (thread_data_t *) thread_param;
+    usleep(thread_func_args->wait_delay*1000);
+    pthread_mutex_lock(thread_func_args->mutex);
+    usleep(thread_func_args->release_delay*1000);
+    pthread_mutex_unlock(thread_func_args->mutex);
+    thread_func_args->thread_complete_success = true;
```

Comment:
@sankalp123321 did the thread complete successfully if each of the steps above failed?